components/user-store/org.wso2.carbon.identity.user.store.count/src/main/java/org/wso2/carbon/identity/user/store/count/jdbc/JDBCUserStoreCountRetriever.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.user.store.count.jdbc;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.user.store.count.AbstractUserStoreCountRetriever;
+import org.wso2.carbon.identity.user.store.count.exception.UserStoreCounterException;
+import org.wso2.carbon.user.api.RealmConfiguration;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.util.DatabaseUtil;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Iterator;
+import java.util.Map;
+
+public class JDBCUserStoreCountRetriever extends AbstractUserStoreCountRetriever {
+
+    private static Log log = LogFactory.getLog(JDBCUserStoreCountRetriever.class);
+    private RealmConfiguration realmConfiguration = null;
+    private int tenantId = 0;
+
+    public JDBCUserStoreCountRetriever() {
+
+    }
+
+    public JDBCUserStoreCountRetriever(RealmConfiguration realmConfiguration, int tenantId) throws UserStoreException {
+        this.realmConfiguration = realmConfiguration;
+        this.tenantId = tenantId;
+    }
```

Comment:
Can we have a init() method instead of this constructor and call the init method. Having this constructor will hide intrinsic details of the API.